main.tf

```diff
@@ -1,11 +1,18 @@
 terraform {
   required_providers {
     cosign = {
-        source = "chainguard-dev/cosign"
+      source = "chainguard-dev/cosign"
     }
   }
+
+  backend "inmem" {}
+
 }
 
+data "cosign_available_credentials" "available" {}
+
 resource "cosign_sign" "image" {
-  image = "us-docker.pkg.dev/wlynch-chainguard/public/ko-gcloud@sha256:d882f1b1ba89f712f00d955c7268d66f89774f79e922258cd6194ae18e8ac7ce"
-}
\ No newline at end of file
+  for_each      = data.cosign_available_credentials.available.available
+  oidc_provider = each.key
```

Comment:
Have you thought about how to use multiple available credentials with the different collision modes?

Reply:
You mean signing with two different GCP SAs? I haven't thought through that yet.

This is just to be able to support signing with `github` + `filesystem` (+ `google` someday) without conflicts.

Reply:
I mean signing with any two ambient identities.  If the `for_each` has >1 entry, how does it work with the replacement modes, which AFAIK are completely agnostic of the signer.